server.js

```diff
@@ -7,20 +7,91 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+const Thought = mongoose.model('Thought', {
+  name: {
+    type: String,
+    default: 'Anonymous',
+    maxlength: 15
+  },
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8041
 const app = express()
 
+const listEndpoints = require('express-list-endpoints')
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+//GET ENDPOINT
+//return 20 results ordered by createAt
+app.get('/thoughts', async (req, res) => {
+  const { page } = req.query
+
+  //pagination
+  const pageNumber = +page || 1
+  const perPage = 20
+  const skip = perPage * (pageNumber - 1)
+
+  const totalThoughts = await Thought.find()
+  const pages = Math.ceil(totalThoughts.length / perPage)
+
+  const thoughts = await Thought.find()
+    .sort({ createdAt: 'desc' })
+    .limit(perPage)
+    .skip(skip)
+
+  res.json({ thoughts, pages })
+})
```

Comment:
according to documentation `get "/"` need to return 20 thoughts,

Reply:
It's paginated so it should only return 20 pages. Is it not working?